buildspec/release/70checkmarketplace.yml

```
@@ -22,21 +22,31 @@ phases:
     build:
         commands:
             - VERSION=$(node -e "console.log(require('./packages/${TARGET_EXTENSION}/package.json').version);")
-            # get extension name
+            # get extension name, also an old version for beta testing
             - |
                 if [ "${TARGET_EXTENSION}" = "amazonq" ]; then
                   extension_name="amazonwebservices.amazon-q-vscode"
+                  beta_version="@1.43.0"
                 elif [ "${TARGET_EXTENSION}" = "toolkit" ]; then
                   extension_name="amazonwebservices.aws-toolkit-vscode"
+                  beta_version="@3.42.0"
                 else
                   echo checkmarketplace: "Unknown TARGET_EXTENSION: ${TARGET_EXTENSION}"
                   exit 1
                 fi
+            # If in beta, install specific extension versions
+            - |
+                if [ "$STAGE" != "prod" ]; then
+                  VERSION=${beta_version#@}
+                  echo "checkmarketplace: Non-production stage detected. Installing beta version '${VERSION}'."
+                else
+                  beta_version=""
+                fi
             # keep reinstalling the extension until the desired version is updated. Otherwise fail on codebuild timeout (1 hour).
             - |
                 while true; do
                   code --uninstall-extension "${extension_name}" --no-sandbox --user-data-dir /tmp/vscode
-                  code --install-extension ${extension_name} --no-sandbox --user-data-dir /tmp/vscode
+                  code --install-extension "${extension_name}${beta_version}" --no-sandbox --user-data-dir /tmp/vscode
```

```suggestion
                  code --install-extension "${extension_name}@${VERSION}" --no-sandbox --user-data-dir /tmp/vscode
```

Always passing the explicit version should work. 

Actually, that might simplify the "validation" a bit too ? Because it will fail if the version doesn't exist.